File: parasmittal/Day 1/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        if(n<=1)
            return 0;
        int profit=0;
        int buy=prices[0];
        for(int i=1;i<n;i++){
            if(buy>prices[i])
                buy=prices[i];
            profit=max(profit,prices[i]-buy);
        }
        return profit;
    }
};

File: sanket-dalvi/Day1/bestTimeToBuyAndSellStock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        if len(prices)<=1:
            return profit
        mn = prices[0]
        for i in range(len(prices)):
            profit = max(profit, prices[i]-mn)
            mn = min(mn, prices[i])
        return profit